### PR TITLE
[RHELC-888] Make RHSM organization and username a secret

### PR DESCRIPTION
**Pull request by bocekm**
Until now we've been recording and collecting the RHSM username and organization in their plain text form in the convert2rhel log files and the breadcrumbs files:
- /var/log/convert2rhel/convert2rhel.log
- /var/log/convert2rhel/archive/convert2rhel-*.log
- /etc/migration-results
- /etc/rhsm/facts/convert2rhel.facts

Users might perceive those values as sensitive or private information. Red Hat doesn't need this information for anything so it's better to not collect it at all.

Jira Issue: [RHELC-888](https://issues.redhat.com/browse/RHELC-888)

Checklist
- [x] PR meets acceptance criteria specified in the Jira issue
- [x] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-888]` is part of the PR title 
- [x] Code and tests are documented properly
- [x] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending`

### CODE REVIEW

File: convert2rhel/breadcrumbs.py
Comment (bocekm) [inline convert2rhel/breadcrumbs.py:96]: No need to have the options to sanitize different for logs and breadcrumbs.
Better to keep the list just on one place.

File: convert2rhel/utils.py
Comment (bocekm) [inline convert2rhel/utils.py:730]: For clarity I've included terminology that made it easier to name variables and form a meaningful debug message.

File: convert2rhel/utils.py
Comment (bocekm) [inline convert2rhel/utils.py:721]: I've removed the "--token" because convert2rhel does not accept such an option and we don't expect it to accept it in the future (I don't know about any kind of token that could be used during the conversion).

**Comment by bocekm** [comment]
TODO: Update related log messages after #681 gets merged (https://github.com/oamg/convert2rhel/pull/681/files#r1096154508).

**Comment by SpyTec** [comment]
#681 is now merged, please rebase

File: convert2rhel/utils.py
Comment (SpyTec) [inline convert2rhel/utils.py:751]: LGTM bot is gone so can remove this
```suggestion
            " secret parameter".format(sanitized_list[-1])
```

**Review by SpyTec — APPROVED** [review]
Failing integration tests unrelated. Can be merged

File: convert2rhel/unit_tests/utils_test.py
Comment (r0x0d) [inline convert2rhel/unit_tests/utils_test.py:724]:
```suggestion
    """Test that the default secret_options are found."""
```

File: convert2rhel/utils.py
Comment (r0x0d) [inline convert2rhel/utils.py:722]: If `-a` were a boolean or a flag of some sort, then having the `-a` in there should be: 

```suggestion
       Example: `ls -a -w 10 dir` ... one option parameter (True, 10)
```

of course, considering argparse in this comment.

Comment (r0x0d) [reply convert2rhel/utils.py:722]: not an suggestion to change, though. Just a comment that I thought it would make sense to add here.

File: convert2rhel/utils.py
Comment (r0x0d) [inline convert2rhel/utils.py:725]: I would write those as `:param:` instead of `:arg:` as it seems more appropriate with the rest of the codebase (at least, the ones who have docstrings that documents the parameters). 

Although, not wrong if they stay in :arg: as we are not using any type of document generation. Up to you to, then.

File: convert2rhel/utils.py
Comment (r0x0d) [inline convert2rhel/utils.py:41]: If you can, document the constant too, please.

**Review by r0x0d — COMMENTED** [review]
Looks good in my opinion, just some small comments that should not block the PR from getting merged.

File: convert2rhel/utils.py
Comment (bocekm) [reply convert2rhel/utils.py:722]: In the terminology I explain how we are to name cli arguments. The "True" value is not an argument passed through CLI. It might be inferred by the CLI utility internally, based on the programming language the utility is written in, or the library used for parsing the CLI arguments.

I'm changing:
```
- an option parameter is the value following an aption if the option requires a value
```
to
```
- an option parameter is the argument following an aption if the option requires a value
```
to make it a bit clearer.

File: tests/integration/tier0/test-check-rollback-handling/test_check_rollback_handling.py
Comment (danmyway) [inline tests/integration/tier0/test-check-rollback-handling/test_check_rollback_handling.py:81]: leftover from #726

**Comment by bocekm** [comment]
The integration test update looks good to me, @danmyway, thanks 👍

**Comment by r0x0d** [comment]
Will give another round of review tomorrow if this was not merged before I start working tomorrow.

File: convert2rhel/utils.py
Comment (r0x0d) [reply convert2rhel/utils.py:752]: That change was not applied, @SpyTec do you want to proceed on merging this and removing the comment later?

**Review by r0x0d — APPROVED** [review]
Still looks good, just one question to @SpyTec if he wants to merge even without his suggestion for removing the lgtm comments was not applied. 

@bocekm one question, though: Should we do the same for the pool parameter? 

Otherwise, feel free to merge whenever you feel like it.

File: convert2rhel/utils.py
Comment (bocekm) [reply convert2rhel/utils.py:752]: Oh, I must have missed that. I can fix it, no problem. We still have a couple other PRs left to finish, so we don't need to merge this now.

File: convert2rhel/utils.py
Comment (bocekm) [reply convert2rhel/utils.py:751]: I dismissed the alert, choosing that it is `False positive`.
Hopefully it's enough so that others won't see this failure in other PRs.

**Comment by bocekm** [comment]
> @bocekm one question, though: Should we do the same for the pool parameter?

No, because Pool IDs are random alphanumerical strings from which no one can infer anything about the subscription, and knowing the string doesn't give those without account credentials the ability to use it in any way.

What we should do though is allowing to pass the username and organization through the new config file. I'll create a task for that.

**Comment by bocekm** [comment]
@danmyway, telling Packit to retest-failed brings unintended consequence these days - all tests are re-run, not just the failed (https://github.com/packit/packit-service/issues/1886). That can lead to a situation where the ones that pass suddenly end up failing :)

Better to re-run the failed tests manually one by one under the [Checks tab](https://github.com/oamg/convert2rhel/pull/725/checks).

**Comment by danmyway** [comment]
> @danmyway, telling Packit to retest-failed brings an unintended consequence these days - all tests are re-run, not just the failed ([packit/packit-service#1886](https://github.com/packit/packit-service/issues/1886)). That can lead to a situation where the ones that previously passed suddenly end up failing :)
> 
> Better to re-run the failed tests manually one by one under the [Checks tab](https://github.com/oamg/convert2rhel/pull/725/checks). That's what I've done now for three failed jobs.

Thanks for the heads-up @bocekm, noted!

**Comment by bocekm** [comment]
@danmyway, FYI I've re-run `testing-farm:oraclelinux-8.6-x86_64:tier1`. There was one test failed due to flaky RHSM.

**Pull request merged**
